### PR TITLE
fix: add rolesWithReviewAuth to config

### LINKED ISSUE #36: Exclude `contributor` reviews in user checks
The checks applied to users when self-assigning need to exclude any `contributor` role based reviews and should only consider:

- `collaborator`
- `member`
- `owner`

_original [context](https://github.com/ubiquity/pay.ubq.fi/pull/290#issuecomment-2333291028)_

### PR DESCRIPTION
**Pull request by Keyrxng**
Resolves #36

### CODE REVIEW

**Comment by gentlementlegen** [comment]
@Keyrxng Can this be reviewed?

**Comment by Keyrxng** [comment]
> @Keyrxng Can this be reviewed?

Yeah it is sorry I forgot to request review. I also didn't provide QA you want me to get some first?

**Comment by Keyrxng** [comment]
@gentlementlegen 

https://github.com/ubq-testing/command-start-stop/issues/20

I left a comment re: my role within my own test org showing as `CONTRIBUTOR` from PR reviews from my main but the weird thing is the comment that I left on the issue (not the review) has my role displayed as member lmao. IDK What is causing the mismatch, I tried to debug but I can't find the cause.

It's not a problem for our prod org or a problem with this PR.

**Comment by gentlementlegen** [comment]
It is quite strange. You are the owner of the organization correct?
Please fix the conflicts so I can review.

**Comment by Keyrxng** [comment]
> You are the owner of the organization correct?

I'm both an `OWNER` and naturally an `ADMIN` too

image

File: src/types/plugin-input.ts [hunk 1]
```diff
@@ -11,14 +11,26 @@ export interface PluginInputs<T extends SupportedEventsU = SupportedEventsU, TU
   ref: string;
 }
 
+const rolesWithReviewAuthority = T.Array(T.String(), { default: ["COLLABORATOR", "OWNER", "MEMBER"] });
```
Comment (gentlementlegen) [inline src/types/plugin-input.ts:14]: Shouldn't `ADMIN` be included here?

File: tests/main.test.ts [hunk 2]
```diff
@@ -93,7 +93,7 @@ describe("User start/stop", () => {
   });
 
   test("Stopping an issue should close the author's linked PR", async () => {
-    const infoSpy = jest.spyOn(console, "info").mockImplementation(() => {});
+    const infoSpy = jest.spyOn(console, "info").mockImplementation(() => { });
```
Comment (gentlementlegen) [inline tests/main.test.ts:96]:
```suggestion
    const infoSpy = jest.spyOn(console, "info").mockImplementation(() => {});
```

File: src/types/plugin-input.ts (hunk 1 above)
Comment (Keyrxng) [reply src/types/plugin-input.ts:14]: Should `COLLABORATOR` be included? Probs not thinking about it

File: tests/main.test.ts [hunk 3]
```diff
@@ -113,7 +113,7 @@ describe("User start/stop", () => {
   });
 
   test("Author's manual unassign should close linked issue", async () => {
-    const infoSpy = jest.spyOn(console, "info").mockImplementation(() => {});
+    const infoSpy = jest.spyOn(console, "info").mockImplementation(() => { });
```
Comment (gentlementlegen) [inline tests/main.test.ts:116]: @Keyrxng did you run prettier on this file and this happened?

Comment (Keyrxng) [reply tests/main.test.ts]: my bad must have forgot to run it after saving

**Review by gentlementlegen — APPROVED** [review]
Seems to work fine on my org. Can I bother you to rename the configuration to `allowedReviewerRoles` to match `automated-merging`?

**Comment by 0x4007** [comment]
3 days is too soon for automated merge

**Pull request merged**